File: python/src/unittest/python/WordDocumentAdapterTest.py
import unittest
from Documents import WordDocument
from Adapters import WordDocumentAdapter
from BaseModule import *


class WordDocumentAdapterTest(unittest.TestCase):

    def setUp(self):
        self.word_doc_adap = WordDocumentAdapter(WordDocument())

    def test_font(self):
        font = self.word_doc_adap.get_font()
        self.assertEqual("Arial", font.get_font_name())
        self.assertEqual(12, font.get_font_size())

    def test_style(self):
        style = self.word_doc_adap.get_style()
        self.assertEqual("Arial", style.get_font().get_font_name())
        self.assertEqual(12, style.get_font().get_font_size())

    def test_background(self):
        background = self.word_doc_adap.get_background()
        self.assertEqual(100.0, background.get_width(), 0.0)
        self.assertEqual(200.0, background.get_height(), 0.0)

    def test_sharing_permissions(self):
        self.assertEqual(1, self.word_doc_adap.get_sharing_permissions())


if __name__ == '__main__':
    unittest.main()

File: java/src/main/test/java/mx/iteso/adapter/WordDocumentAdapterTest.java
package mx.iteso.adapter;

import mx.iteso.adapter.WordDocument;
import mx.iteso.adapter.WordDocumentAdapter;
import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class WordDocumentAdapterTest {
    private WordDocumentAdapter adapter;

    @Before
    public void init() {
        adapter = new WordDocumentAdapter(new WordDocument());
    }

    @Test
    public void getFontTest() {
        Font font = new Font("TimesRoman", Font.PLAIN, FONT_SIZE);
        assertEquals(font.getFontName(),googleDoc.getFont().getFontName());
    }

    @Test
    public void getBackgroundTest() {
        assertEquals("image.png",adapter.getBackground().getPath());
        assertEquals(1024.0f,adapter.getBackground().getWidth());
        assertEquals(720.0f,adapter.getBackground().getHeight());
    }
    @Test
    public void getStyleTest() {
        Format style = new Format("Arial", 12);
        assertEquals(style,adapter.getStyle());
    }
    @Test
    public void getSharingPermissionsTest() {
        adapter.setSharingPermissions(1);
        assertEquals(1,googleDoc.getSharingPermissions());
    }
}